partner_pricelist_search/models/res_partner.py

```diff
@@ -12,6 +12,20 @@ class Pricelist(models.Model):
         search="_search_property_product_pricelist"
     )
 
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        # Substitute pricelist tuple
+        partner_domain = [
+            (1, "=", 1)
+            if (isinstance(x, tuple) and x[0] == "property_product_pricelist")
+            else x
+            for x in args
+        ]
+        self.env.context = dict(self.env.context, search_partner_domain=partner_domain)
```

Comment:
I understand that if you do this it's because you can't use with_context, right?
I don't know why I thought the context was "frozen" in this version.